Editors/Audio/Presentation/AudioEditor/ViewModels/AudioEditorViewModel.cs

```diff
@@ -0,0 +1,291 @@
+﻿using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Windows.Input;
+using CommonControls.PackFileBrowser;
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+using Editors.Audio.Storage;
+using Newtonsoft.Json;
+using Shared.Core.Misc;
+using Shared.Core.PackFiles;
+using Shared.Core.PackFiles.Models;
+using Shared.Core.ToolCreation;
+using static Editors.Audio.Presentation.AudioEditor.AudioEditorViewModelHelpers;
+
+namespace Editors.Audio.Presentation.AudioEditor.ViewModels
+{
+    public class CustomStateDataGridProperties
+    {
+        public string CustomVOActor { get; set; }
+        public string CustomVOCulture { get; set; }
+        public string CustomVOBattleSelection { get; set; }
+        public string CustomVOBattleSpecialAbility { get; set; }
+        public string CustomVOFactionLeader { get; set; }
+    }
+
+    public enum DialogueEventsPreset
+    {
+        None,
+        All,
+        Essential,
+        Custom
+    }
+
+    public partial class AudioEditorViewModel : ObservableObject, IEditorViewModel
+    {
+        public readonly IAudioRepository AudioRepository;
+        public readonly PackFileService PackFileService;
+
+        public string _previousSelectedAudioProjectEvent;
+        [ObservableProperty]
```

Comment:
preference, but I think it looks very neat when you do this as one line
[observableProp] float _someAttribute

Reply:
Didn't even think of that, nice will do.